ibis/backends/dask/execution/selection.py

```diff
@@ -130,7 +131,21 @@ def build_df_from_projection(
     Build up a df from individual pieces by dispatching to `compute_projection`
     for each expression.
     """
+    # Fast path for when we're assigning columns into the same table.
+    if (selection_exprs[0] is op.table) and all(
+        is_row_order_preserving(selection_exprs[1:])
+    ):
+        for expr in selection_exprs[1:]:
+            projection = compute_projection(expr, op, data, **kwargs)
+            if isinstance(projection, dd.Series):
+                data = data.assign(**{projection.name: projection})
+            else:
+                data = data.assign(
+                    **{c: projection[c] for c in projection.columns}
```

Comment:
it might be better if we have a *lot* of columns to use the concat path, though would require a heurestic / magic number e.g. say > 100 or something

Reply:
Seems like a good time to kick the tires on the new config stuff!

Reply:
This can be a follow up if it ever becomes an issue!